hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/wal/FSHLog.java

```diff
@@ -390,6 +390,9 @@ protected void doReplaceWriter(Path oldPath, Path newPath, Writer nextWriter) th
             closeWriter(this.writer, oldPath, true);
           } finally {
             inflightWALClosures.remove(oldPath.getName());
+            if (!isUnflushedEntries()) {
```

Comment:
Similar to the code below at line 403, should we also have a catch block to Log a WARNING?
Does it make any difference to first remove old path and then call markCloseAndClean or vice versa? Since the JIRA said we are trying to follow the similar as below, we can try to keep it same?

Reply:
Just another question, does closeWriter also flush?

Reply:
@ranganathg `inflightWALClosures.remove(oldPath.getName());` was there previously as well so not changing the behaviour there. We are explicitly looking for the closing for cleanup

>Just another question, does closeWriter also flush?
No, this is in the write path it will not trigger any flush

Reply:
@ranganathg sorry I got confused with the flush for region. If you are asking for flush for the entries that are there on the ringbuffer that will be done by the sycn runner threads which will increment the sequenceId